src/dialects/abstract/query-generator.d.ts

```diff
@@ -166,6 +173,11 @@ export class AbstractQueryGenerator {
     options?: ArithmeticQueryOptions,
   ): string;
 
+  showIndexesQuery(tableName: TableName): string;
+
+  dropTableQuery(tableName: TableName, options?: DropTableQueryOptions): string;
+  describeTableQuery(tableName: TableName, schema?: string, schemaDelimiter?: string): string;
```

Comment:
`schemaDelimiter` is only supported in sqlite, do we want to keep this or drop this option?

Reply:
The current design of `describeTableQuery` makes no sense to me. `TableName` is already capable of specifying a schema according to its definition:

```ts
export interface TableNameWithSchema {
  tableName: string;
  schema?: string;
  delimiter?: string;
  as?: string;
  name?: string;
}
export type TableName = string | TableNameWithSchema;
```

So this method doesn't need extra parameters, just a single `TableName` one:

```ts
describeTableQuery('users')
describeTableQuery({ tableName: 'users', schema: 'public' })
```

This one is inconsistent and should be removed (breaking change)

```ts
describeTableQuery('users', 'public')
```

Reply:
Yeah, that's something I noticed yesterday when we discussed the `extractTableDetails` issue. I think I'll split parts of this PR to another where we use the `TableName` type consistenly and not allow schema/delimiter to be passed seperately.

Reply:
Sounds good!